sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/tests/ScenarioTests/SqlResourcesOperationsTests.cs

```diff
@@ -264,6 +265,418 @@ public void SqlCRUDTests()
+                            RestoreTimestampInUtc = restoreTimestampInUtc
+                        },
+                        CreateMode = CreateMode.Restore
+                    }
+                };
+                SqlContainerGetResults sqlContainerGetResults2 = client.CreateUpdateSqlContainerWithHttpMessagesAsync(
+                    this.fixture.ResourceGroupName,
+                    databaseAccountName,
+                    databaseName,
+                    containerName,
+                    sqlContainerCreateUpdateParameters2).GetAwaiter().GetResult().Body;
+
+                Assert.NotNull(sqlContainerGetResults2);
+                VerifyEqualSqlContainers(sqlContainerGetResults, sqlContainerGetResults2);
+
+                //delete database
+                await client.DeleteSqlDatabaseWithHttpMessagesAsync(this.fixture.ResourceGroupName, databaseAccountName, databaseName);
+
+                //restore database
+                SqlDatabaseCreateUpdateParameters sqlDatabaseCreateUpdateParameters3 = new SqlDatabaseCreateUpdateParameters
+                {
+                    Resource = new SqlDatabaseResource
+                    {
+                        Id = databaseName,
+                        RestoreParameters = new ResourceRestoreParameters
+                        {
+                            RestoreSource = restoreSource,
+                            RestoreTimestampInUtc = restoreTimestampInUtc
+                        },
+                        CreateMode = CreateMode.Restore
+                    }
+                };
+
+                SqlDatabaseGetResults sqlDatabaseGetResults4 = client.CreateUpdateSqlDatabaseWithHttpMessagesAsync(
+                    this.fixture.ResourceGroupName,
+                    databaseAccountName,
+                    databaseName,
+                    sqlDatabaseCreateUpdateParameters3
+                ).GetAwaiter().GetResult().Body;
+                Assert.NotNull(sqlDatabaseGetResults4);
+                VerifyEqualSqlDatabases(sqlDatabaseGetResults, sqlDatabaseGetResults4, true);
+
+                clientEncryptionKeyRetrieved = client.GetClientEncryptionKeyWithHttpMessagesAsync(
+                    this.fixture.ResourceGroupName,
+                    databaseAccountName,
+                    databaseName,
+                    clientEncryptionKeyName1).GetAwaiter().GetResult().Body;
+                Assert.NotNull(clientEncryptionKeyRetrieved);
+                Assert.Equal(clientEncryptionKeyResource.Id, clientEncryptionKeyRetrieved.Resource.Id);
+                Assert.Equal(clientEncryptionKeyResource.EncryptionAlgorithm, clientEncryptionKeyRetrieved.Resource.EncryptionAlgorithm);
+                Assert.Equal(clientEncryptionKeyResource.KeyWrapMetadata.Name, clientEncryptionKeyRetrieved.Resource.KeyWrapMetadata.Name);
+                //Assert.Equal(clientEncryptionKeyResource.KeyWrapMetadata.Algorithm, clientEncryptionKeyRetrieved.Resource.KeyWrapMetadata.Algorithm);
```

Comment:
this is fixed so we can validate by enabling this. Not sure, if the fix has been ported to 0601